tests/functional/src/main/java/org/mule/functional/junit4/ArtifactFunctionalTestCase.java

```diff
@@ -0,0 +1,72 @@
+ */
+
+package org.mule.functional.junit4;
+
+import static org.mule.functional.util.AnnotationUtils.getAnnotationAttributeFrom;
+import org.mule.functional.classloading.isolation.builder.ClassLoaderIsolatedExtensionsManagerConfigurationBuilder;
+import org.mule.functional.junit4.runners.ArtifactClassLoaderRunnerConfig;
+import org.mule.functional.junit4.runners.ArtifactClassloaderTestRunner;
+import org.mule.functional.junit4.runners.PluginClassLoadersAware;
+import org.mule.functional.junit4.runners.RunnerDelegateTo;
+import org.mule.runtime.core.api.config.ConfigurationBuilder;
+import org.mule.runtime.module.artifact.classloader.ArtifactClassLoader;
+
+import java.util.List;
+
+import org.junit.runner.RunWith;
+
+/**
+ * Base class for running {@link FunctionalTestCase} with class loader isolation that uses a {@link org.junit.runner.Runner}
+ * that classifies the classpath provided by IDE/surfire-maven-plugin and classifies the URLs using the maven dependency graph.
+ * <p/>
+ * It has support for extensions and if the test class is annotated with {@link ArtifactClassLoaderRunnerConfig} and
+ * extensions classes are defined.
+ * <p/>
+ * By default this test runs internally with a {@link org.junit.runners.JUnit4} runner.
+ * For those cases where the test has to be run with another runner the {@link RunnerDelegateTo} should be used to define it.
+ *
+ * @since 4.0
+ */
+@RunWith(ArtifactClassloaderTestRunner.class)
+public abstract class ArtifactFunctionalTestCase extends FunctionalTestCase
+{
+    private static List<ArtifactClassLoader> extensionsClassLoaders;
+
+    /**
+     * @return the TCCL as it will be the application {@link ClassLoader} created by the runner.
+     */
+    @Override
+    protected ClassLoader getExecutionClassLoader()
+    {
+        return Thread.currentThread().getContextClassLoader();
+    }
+
+    @PluginClassLoadersAware
+    public static void setExtensionClassLoaders(List<ArtifactClassLoader> extensionPluginsArtifactClassLoaders)
+    {
+        extensionsClassLoaders = extensionPluginsArtifactClassLoaders;
+    }
+
+    @Override
+    protected void addBuilders(List<ConfigurationBuilder> builders)
+    {
+        super.addBuilders(builders);
+        Class<?> runner = getAnnotationAttributeFrom(this.getClass(), RunWith.class, "value");
+        if (runner == null || !runner.equals(ArtifactClassloaderTestRunner.class))
+        {
+            throw new IllegalStateException(this.getClass().getName() + " extends " + ArtifactFunctionalTestCase.class.getName()
+                                            + " so it should be annotated to only run with: " + ArtifactClassloaderTestRunner.class + ". See " + RunnerDelegateTo.class + " for defining a delegate runner to be used.");
+        }
+
+        if (extensionsClassLoaders != null && !extensionsClassLoaders.isEmpty())
+        {
+            builders.add(0, new ClassLoaderIsolatedExtensionsManagerConfigurationBuilder(extensionsClassLoaders));
```

Comment:
Copied from ExtensionFunctionalTestCase and it has to be added into the first place if not it fails :S
Caused by: org.springframework.beans.factory.parsing.BeanDefinitionParsingException: Configuration problem: The element 'config' does not have an associated Bean Definition Parser.  Is the module or extension associated with http://www.mulesoft.org/schema/mule/file present on the classpath?
Offending resource: URL [file:/Users/xxx/Development/mule/extensions/file/target/test-classes/file-read-config.xml]
	at org.springframework.beans.factory.parsing.FailFastProblemReporter.fatal(FailFastProblemReporter.java:60)

And actually the file is there.